### PR TITLE
Implented Gen VII Sheer Cold Changes

### PR DESCRIPTION
**Pull request by EvasiveAce**
Gen VII added two things to sheer cold:
- _Ice-type Pokémon are now immune to Sheer Cold._
- _Additionally, the move's base chance of hitting is only 20% if the user is not an Ice-type Pokémon._

I tested it with increased accuracy for the ice immunity. 
I also split the attributes into two, one for accuracy and one for Ice immunity.

https://bulbapedia.bulbagarden.net/wiki/Sheer_Cold_(move)

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -4625,9 +4653,9 @@ export function initMoves() {
     new AttackMove(Moves.SAND_TOMB, Type.GROUND, MoveCategory.PHYSICAL, 35, 85, 15, 100, 0, 3)
       .attr(TrapAttr, BattlerTagType.SAND_TOMB)
       .makesContact(false),
-    new AttackMove(Moves.SHEER_COLD, Type.ICE, MoveCategory.SPECIAL, 200, 30, 5, -1, 0, 3)
-      .attr(OneHitKOAttr)
-      .attr(OneHitKOAccuracyAttr),
+      new AttackMove(Moves.SHEER_COLD, Type.ICE, MoveCategory.SPECIAL, 200, 20, 5, -1, 0, 3)
```
Comment (TempsRay) [inline src/data/move.ts:4656]: nit: misplaced tabbing

File: src/data/move.ts [hunk 2]
```diff
@@ -2329,6 +2329,17 @@ export class WaterSuperEffectTypeMultiplierAttr extends VariableMoveTypeMultipli
   }
 }
 
+export class IceNoEffectTypeMultiplierAttr extends VariableMoveTypeMultiplierAttr {
```
Comment (TempsRay) [inline src/data/move.ts:2332]: Isn't Sheer Cold completely missing the OHKO implementation now?

File: src/data/move.ts [hunk 3]
```diff
@@ -2329,6 +2329,17 @@ export class WaterSuperEffectTypeMultiplierAttr extends VariableMoveTypeMultipli
...
+   apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
+    const multiplier = args[0] as Utils.NumberHolder;
+    if (target.isOfType(Type.ICE)) {
+      multiplier.value *= 0; // Increased twice because initial reduction against water
```
Comment (TempsRay) [inline src/data/move.ts:2336]: nit: I don't quite get the comment here Why not just set it to 0?

**Review by TempsRay — COMMENTED** [review]
Looks good to me for the most part. The only thing I'm wondering is if the `IceNoEffectTypeMultiplierAttr` setting the damage to 0 is equivalent to the entire move failing.

**Review by TempsRay — APPROVED** [review]

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**